errors/either/payment-done.sc

```diff
@@ -1,16 +1,31 @@
-type UserId = String
-def getUser(email: String): Either[String,UserId] = ???
-def createUser(email: String): Either[String,UserId] = ???
+object UserStore {
+  type UserId = String
 
-def executePayment(amount: Int, userId: String): Either[String,Unit] = ???
+  def getUser(email: String): Either[String,UserId] = if (email == "a@gu.com") Right("1") else Left("Not found")
+  def createUser(email: String): Either[String,UserId] = Right("2")
+}
+
+def validateEmail(email: String): Either[String,String] = if (email.contains("@")) Right(email) else Left("Invalid email")
+
+def executePayment(amount: Int, userId: String): Either[String,Unit] = if (amount > 0) Right(()) else Left("Invalid amount")
 
 case class Request(amount: Int, email: String)
 type HttpStatus = Int
 
+/**
+ * 1. validate the email
+ * 2. get or create a user ID
+ * 3. execute the payment
+ * 4. return a 200 status on success
+ */
 def handleRequest(request: Request): HttpStatus = {
```

Comment:
let's call this `processContribution`. or something similar on those lines, i.e. we're pretending this is to execute one-off contribution for a specific user. `Request` type can be called `OneOffContribution`. makes it like a real-world Guardian use case, and means we don't have a generic name like `handleRequest`